### PR TITLE
Feature/auto assign

### PR DESCRIPTION
**Pull request by Tarrke**
Here is the implementation of Issue #28.

This has been made an option and is working right out the box on my old save.

I made the placeholder for the german translation, fixed the french one, and tryed to make an english locale looking good.

### CODE REVIEW

**Comment by Tarrke** [comment]
Jonas could you review this PR?

File: src/todo/UI.lua [hunk 1]
```diff
@@ -145,13 +145,24 @@ function todo.create_add_edit_frame(player)
...
-        selected_index = 1
-    })
+    local players, _, c = todo.get_player_list()
+    -- The count is off by 1 so if the result is 2 we get only one player
+    if( todo.auto_assign(player) and c == 2 ) then
+        table.add({
+            type = "drop-down",
+            name = "todo_add_assignee_drop_down",
+            items = players,
+            selected_index = 2
```
Comment (JonasJurczok) [inline src/todo/UI.lua]: I would only do the index decision in the if and then use the resulting value in the add block. This makes it a little bit more clear where the difference is and what is actually decided.

File: src/todo/helper.lua [hunk 2]
```diff
@@ -13,20 +13,23 @@ function todo.log(message)
...
 function todo.get_player_list()
-    local result = {{"todo.unassigned"} }
+    local result = { {"todo.unassigned"} }
 
     for _, player in pairs(game.players) do
         table.insert(result, player.name)
     end
 
     local lookup = {}
+    local count = 0
```
Comment (JonasJurczok) [inline src/todo/helper.lua]: You could fix the off by one error by doing the counting in the loop above. Then you get the real result.

File: src/todo/helper.lua [hunk 3]
```diff
@@ -61,6 +64,10 @@ function todo.show_minimized(player)
     return settings.get_player_settings(player)["todolist-show-minimized"].value
 end
 
+function todo.auto_assign(player)
```
Comment (JonasJurczok) [inline src/todo/helper.lua]: You could rename this to "is_auto_assign" or "get_auto_assign" to make it clearer that this function is not actually assigning anything but just reading the settings.

File: src/todo/helper.lua (hunk 2 above)
Comment (Tarrke) [reply src/todo/helper.lua]: I have to trust you on that one. I'll check this in game tonight if I catch the time to. Why is it different in the two loops?

**Comment by Tarrke** [comment]
I did the changes quickly this morning. I'll double check thoses tonight.

File: src/todo/UI.lua [hunk 4]
```diff
@@ -145,13 +145,20 @@ function todo.create_add_edit_frame(player)
...
-        selected_index = 1
-    })
+        items = players
+    }
+    if( todo.is_auto_assign(player) and c == 1 ) then
+        t.selected_index = 2
+    else
+        t.selected_index = 1
+    end
+    table.add(t)
```
Comment (JonasJurczok) [inline src/todo/UI.lua]: What I meant is more like this
```
local index = 0
if (.) then
index = 1
else
index = 2
end

table.add({
...
selected_index = index
})
````

Then you don't have to mix execution and details that much.

File: src/todo/helper.lua [hunk 5]
```diff
@@ -25,8 +27,9 @@ function todo.get_player_list()
     end
 
     todo.log("Created Assignee list: " .. serpent.block(result))
+    todo.log("Players in the game : " .. count - 1)
```
Comment (JonasJurczok) [inline src/todo/helper.lua]: It shouldn't be off by one anymore :)

File: src/todo/UI.lua [hunk 6]
```diff
@@ -145,13 +145,20 @@ function todo.create_add_edit_frame(player)
     })
 
 
-    local players, _ = todo.get_player_list()
-    table.add({
+    local players, _, c = todo.get_player_list()
+    -- The count is off by 1 so if the result is 2 we get only one player
```
Comment (JonasJurczok) [inline src/todo/UI.lua]: This should not be true anymore :)

File: src/todo/UI.lua (hunk 4 above)
Comment (Tarrke) [reply src/todo/UI.lua]: Then something like:
```lua
local index = 2
if( . ) then
  index = 1
end
table.add({
....
selected_index = index
})
```

**Comment by Tarrke** [comment]
This version should be good.

File: src/todo/UI.lua [hunk 7]
```diff
@@ -145,14 +145,19 @@ function todo.create_add_edit_frame(player)
     })
 
 
-    local players, _ = todo.get_player_list()
+    local players, _, c = todo.get_player_list()
+    local index = 1
+    if( todo.is_auto_assign(player) and c == 1 ) then
```
Comment (JonasJurczok) [inline src/todo/UI.lua]: yep, nice.

File: src/todo/UI.lua [hunk 8]
```diff
@@ -145,14 +145,19 @@ function todo.create_add_edit_frame(player)
...
+    end
     table.add({
         type = "drop-down",
         name = "todo_add_assignee_drop_down",
         items = players,
-        selected_index = 1
+        selected_index = index
     })
 
+
```
Comment (JonasJurczok) [inline src/todo/UI.lua]: blank line

**Comment by JonasJurczok** [comment]
Nice. I can't modify your branch so if you could bump the version to 16.1.0 in info.json then I can release it directly. 

Thanks again :)

**Comment by Tarrke** [comment]
OK done and done. I hope everything's fine.

**Pull request merged**